rustler_mix/lib/rustler/compiler.ex

```diff
@@ -122,13 +122,47 @@ defmodule Rustler.Compiler do
   defp make_build_mode_flag(args, :release), do: args ++ ["--release"]
   defp make_build_mode_flag(args, :debug), do: args
 
+  defp get_target_os_type(nil) do
+    :os.type()
+  end
+
+  defp get_target_os_type(target) when is_binary(target) do
+    os_type =
+      [
+        %{pattern: ~r/.*-linux.*/, os_type: {:unix, :linux}},
+        %{pattern: ~r/.*-windows.*/, os_type: {:win32, :nt}},
+        %{pattern: ~r/.*-apple.*/, os_type: {:unix, :darwin}},
+        %{pattern: ~r/.*-freebsd.*/, os_type: {:unix, :freebsd}},
+        %{pattern: ~r/.*-netbsd.*/, os_type: {:unix, :netbsd}},
+        %{pattern: ~r/.*-openbsd.*/, os_type: {:unix, :openbsd}},
+        %{pattern: ~r/.*-solaris.*/, os_type: {:unix, :solaris}}
+      ]
+      |> Enum.map(&%{matched: Regex.match?(&1.pattern, target), os_type: &1.os_type})
+      |> Enum.reject(&(&1.matched == false))
```

Comment:
As we are only interested in a single match, you can use `Enum.find/2`:

```elixir
os_type = ...
|> Enum.map(...)
|> Enum.find(& &1.matched))

if os_type, do: os_type, else: raise
```

Reply:
Oh, thank you for the suggestion! I didn't know about this function. I'll update the code.